src/ziggurat/messaging/producer.clj

```diff
@@ -103,9 +103,20 @@
   "Get exponential backoff enabled for specific channel from config."
   (-> (ziggurat-config) :stream-router topic-entity :channels channel :retry :exponential-backoff-enabled))
 
-(defn- get-exponential-backoff [retry-count message-retry-count queue-timeout-ms]
+(defn- get-exponential-backoff [topic-entity channel message-payload]
+  "Get exponential-backoff value when exponential backoff is enabled."
+  (let [retry-count (-> (ziggurat-config) :retry :count)
+        channel-retry-count (get-channel-retry-count topic-entity channel)
+        message-retry-count (:retry-count message-payload)
+        exponent (- (or channel-retry-count retry-count) message-retry-count)]
+    (if (< exponent 10)
```

Comment:
why have we capped this off at 10. And please add this in the documentation as well.

Reply:
this is to avoid to many channel, can make this configurable but for current use case max 10 exponent should be enough.

Reply:
updated with documentation

Reply:
this is updated, move maximum exponential retry count to config and get-backoff-exponent now accept :message-retry-count